### PR TITLE
[BugFix] fix shadow partition show or not

### LINKED ISSUE #26996: [auto partition staros] execute "show partitions" result is incorrect
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
create database auto_set_bucket_3976b5e4_1fdd_11ee_9d75_00163e0e550b;
use auto_set_bucket_3976b5e4_1fdd_11ee_9d75_00163e0e550b
 CREATE TABLE `duplicate_table_with_null` ( `k1`  date, `k2`  datetime,`k3`  char(20), `k4`  varchar(20), `k5`  boolean, `k6`  tinyint, `k7`  smallint, `k8`  int, `k9`  bigint, `k10` largeint, `k11` float, `k12` double, `k13` decimal(27,9)) DUPLICATE KEY(`k1`, `k2`, `k3`, `k4`, `k5`) PARTITION BY time_slice(k2, interval 1 day) DISTRIBUTED BY HASH(`k1`, `k2`, `k3`) PROPERTIES (  "replication_num" = "3", "storage_format" = "v2");

 LOAD LABEL auto_set_bucket_3976b5e4_1fdd_11ee_9d75_00163e0e550b.load_3d77e2c6_1fdd_11ee_9d75_00163e0e550b (DATA INFILE( "hdfs://xxx.xx.xxx.xxx:xxxx/starrocks_test_data/system_data/hdfs_data/nose-ut-data/data") INTO TABLE `duplicate_table_with_null` WHERE k1 is not null and k1 > '2020-07-31' and  k1 < '2020-08-05')  WITH BROKER ("username"="sr","password"="sr@test") PROPERTIES("timeout"="300");

 show partitions from duplicate_table_with_null order by PartitionName;
```

### Expected behavior (Required)
no $shadow_automatic_partition
image

### Real behavior (Required)
has $shadow_automatic_partition
image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
mysql> select current_version();
+--------------------+
| current_version()  |
+--------------------+
| branch-3.1-5a3acb9 |
+--------------------+
1 row in set (0.01 sec)

### PR DESCRIPTION
**Pull request by kevincai**
* unify olap table show partition operation
* fixes #26996

Fixes #26996 

## What type of PR is this:
- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|6|8|75.00%|[279, 308]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|6|8|75.00%|[279, 308]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 8 (75.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|6|8|75.00%|[279, 308]|

**Comment by kevincai** [comment]
sql tester failure due to concurrent running cases. non related to this PR change.

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java [hunk 1]
```diff
@@ -264,13 +265,19 @@ public BaseProcResult getBasicProcResult(List<List<Comparable>> partitionInfos)
     }
 
     private List<List<Comparable>> getPartitionInfos() {
-        return table.isCloudNativeTableOrMaterializedView() ? getLakePartitionInfos() : getOlapPartitionInfos();
+        return getOlapPartitionInfos();
     }
 
     private List<List<Comparable>> getOlapPartitionInfos() {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java:271]: line 268 - 271 can be removed

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/PartitionsProcDir.java]: done

**Review by Astralidea — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 7 (71.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|5|7|71.43%|[275, 304]|

**Pull request merged**